node/pkg/watchers/evm/ccq.go

```diff
@@ -211,6 +211,61 @@ func (w *Watcher) ccqHandleEthCallByTimestampQueryRequest(logger *zap.Logger, ct
 		zap.Int("numRequests", len(req.CallData)),
 	)
 
+	if (block == "") != (nextBlock == "") {
+		logger.Error("invalid block id hints in eth_call_by_timestamp query request, if one is unset they both must be unset",
+			zap.Uint64("timestamp", req.TargetTimestamp),
+			zap.String("block", block),
+			zap.String("nextBlock", nextBlock),
+		)
+		w.ccqSendQueryResponseForError(logger, queryRequest, query.QueryFatalError)
+		return
+	}
+
+	if block == "" {
+		if w.ccqTimestampCache == nil {
+			logger.Error("error in block id hints in eth_call_by_timestamp query request, they are unset and chain does not support timestamp caching")
+			w.ccqSendQueryResponseForError(logger, queryRequest, query.QueryFatalError)
+			return
+		}
+
+		// Look the timestamp up in the cache. Note that the cache uses native EVM time, which is seconds, but CCQ uses milliseconds, so we have to convert.
+		blockNum, nextBlockNum, found := w.ccqTimestampCache.LookUp(req.TargetTimestamp / 1000000)
+		if !found {
+			if blockNum == 0 {
```

Comment:
this is also be the case when the cache is empty though, right? which might be helped by waiting? guess that would only be a startup issue so it's likely not a big deal and would only be for the first second of the watcher lifecycle

Reply:
Changed it to retry.